task/sql-tasks.js

```diff
@@ -280,30 +461,49 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+  let result = await db.query(`
+        SELECT DISTINCT
+            C.CompanyName,
+            P.ProductName,
+            OD.UnitPrice as "PricePerItem"
+        FROM Customers as C
+        JOIN Orders as O ON O.CustomerID = C.CustomerID        
+        JOIN OrderDetails as OD ON OD.OrderID = O.OrderID
+        JOIN Products as P ON OD.ProductID = P.ProductID
+        JOIN (SELECT
+          CNew.CustomerID,
+          MAX(ODNew.UnitPrice) as 'MaxPrice'
+          FROM Customers as CNew
+          JOIN Orders as ONew ON ONew.CustomerID = CNew.CustomerID 
+          JOIN OrderDetails as ODNew ON ONew.OrderID = ODNew.OrderID 
+          GROUP BY CNew.CustomerID) as S
+          ON S.CustomerID = C.CustomerID AND S.MaxPrice = OD.UnitPrice
+          ORDER BY OD.UnitPrice DESC, C.CompanyName, P.ProductName;
+  `);
+  return result[0];
 }
 
 module.exports = {
-    task_1_1: task_1_1,
-    task_1_2: task_1_2,
-    task_1_3: task_1_3,
-    task_1_4: task_1_4,
-    task_1_5: task_1_5,
-    task_1_6: task_1_6,
-    task_1_7: task_1_7,
-    task_1_8: task_1_8,
-    task_1_9: task_1_9,
-    task_1_10: task_1_10,
-    task_1_11: task_1_11,
-    task_1_12: task_1_12,
-    task_1_13: task_1_13,
-    task_1_14: task_1_14,
-    task_1_15: task_1_15,
-    task_1_16: task_1_16,
-    task_1_17: task_1_17,
-    task_1_18: task_1_18,
-    task_1_19: task_1_19,
-    task_1_20: task_1_20,
-    task_1_21: task_1_21,
-    task_1_22: task_1_22
+  task_1_1: task_1_1,

```

Comment:
github shows that this lines are changed. Do not modify original code

![image](https://user-images.githubusercontent.com/7272208/92419745-04602200-f178-11ea-9238-6b62c786bbf8.png)